tests/test_vault.py

```diff
@@ -0,0 +1,162 @@
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"vault secrets tests"
+
+import os
+import unittest
+import tempfile
+import base64
+import hvac
+import docker
+import shutil
+from time import sleep
+from kapitan.refs.base import RefController, RefParams, Revealer
+from kapitan.refs.secrets.vaultkv import  vault_obj, VaultSecret
+
+# Create temporary folder
+REFS_HOME = tempfile.mkdtemp()
+REF_CONTROLLER = RefController(REFS_HOME)
+REF_CONTROLLER.target = 'minikube-mysql'
+REF_CONTROLLER.inventory = os.path.join('.','examples','kubernetes','inventory')
+REVEALER = Revealer(REF_CONTROLLER)
+# Create Vault docker container
+client = docker.from_env()
+env = {'VAULT_LOCAL_CONFIG':'{"backend": {"file": {"path": "/vault/file"}}, "listener":{"tcp":{"address":"0.0.0.0:8200","tls_disable":"true"}}}'}
+vault_container= client.containers.run(image='vault',cap_add=['IPC_LOCK'],ports={'8200':'8200'},
+                                       environment=env,detach=True,remove=True,command='server')
+sleep(2)
+
+class VaultSecretTest(unittest.TestCase):
+    "Test Vault Secret"
+
+    @classmethod
+    def setUpClass(cls):
+        # Initialize vault, unseal, mount secret engine & add auth
+        cls.client = hvac.Client()
+        init = cls.client.sys.initialize()
+        cls.client.sys.submit_unseal_keys(init['keys'])
+        os.environ['VAULT_ROOT_TOKEN'] = init['root_token']
+        cls.client.adapter.close()
+        cls.client = hvac.Client(token = init['root_token'])
+        cls.client.sys.enable_secrets_engine(backend_type='kv-v2',path='secret')
+        test_policy = '''
+        path "secret/*" {
+          capabilities = ["read", "list"]
+        }
+        '''
+        policy = 'test_policy'
+        cls.client.sys.create_or_update_policy(name=policy,policy=test_policy)
+        os.environ['VAULT_USERNAME'] = 'test_user'
+        os.environ['VAULT_PASSWORD'] = 'test_password'
+        cls.client.sys.enable_auth_method('userpass')
+        cls.client.create_userpass(username='test_user', password='test_password', policies=[ policy ])
+        cls.client.sys.enable_auth_method('approle')
+        cls.client.create_role('test_role')
+        os.environ['VAULT_ROLE_ID'] = cls.client.get_role_id('test_role')
+        os.environ['VAULT_SECRET_ID'] = cls.client.create_role_secret_id('test_role')['data']['secret_id']
+        os.environ['VAULT_TOKEN'] = cls.client.create_token(policies=[policy],lease='1h')['auth']['client_token']
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.client.adapter.close()
+        vault_container.stop()
+        shutil.rmtree(REFS_HOME, ignore_errors=True)
```

Comment:
Remember to clear `VAULT_ROOT_TOKEN` and all the other env vars to "" after tests run.